src/quantum/azext_quantum/commands.py

```diff
@@ -0,0 +1,104 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long
+
+from collections import OrderedDict
+from azure.cli.core.commands import CliCommandType
+from ._validators import validate_workspace_info, validate_target_info, validate_workspace_and_target_info
+
+
+def transform_targets(providers):
+    def one(provider, target):
+        return OrderedDict([
+            ('Provider', provider),
+            ('Target-id', target['id']),
+            ('Status', target['currentAvailability']),
+            ('Average Queue Time', target['averageQueueTime'])
+        ])
+
+    targets = []
+    for provider in providers:
+        for target in provider['targets']:
+            targets.append(one(provider['id'], target))
+    return targets
```

Comment:
Minor, but it would be slightly more Python-ic to make this a list comprehension:
```suggestion
    targets = [
        one(provider['id'], target)
        for provider in providers
        for target in provider['targets']
    ]
``` #Resolved